osu.Game/Rulesets/Edit/IPlayfieldProvider.cs

```diff
@@ -1,11 +1,12 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
-namespace osu.Game.Rulesets.Mania.Edit.Blueprints
+using osu.Game.Rulesets.UI;
+
+namespace osu.Game.Rulesets.Edit
 {
-    public enum HoldNotePosition
+    public interface IPlayfieldProvider
```

Comment:
I don't believe this interface is necessary to exist. An alternative way that I tested to work goes as follows:

* `ManiaSelectionBlueprintTestScene` is the easy case, as its `Playfield` can just have `[Cached(typeof(Playfield))]` on it.
* As for `HitObjectComposer<T>`, it's a bit trickier, but this can be done without the interface by overriding `CreateChildDependencies()`, creating a mutable dependency container there and storing it to a field, and then caching the `Playfield` in BDL. The [mania composer](https://github.com/ppy/osu/blob/044ff9d1d274cdc4ac1903c79064e97e3178c448/osu.Game.Rulesets.Mania/Edit/ManiaHitObjectComposer.cs#L85-L86) even does something like that already with `IScrollingInfo`.

Whether that's better, I'm not entirely sure, but it works without the interface and I'd personally prefer not to have it I think.

Reply:
I didn't know that way of using the dependency container. I appreciate.